sourcegen-model/src/main/java/io/micronaut/sourcegen/model/MethodDef.java

```diff
@@ -186,8 +186,11 @@ public MethodDef build() {
             return build((self, parameterDefs) -> null);
         }
 
-        public MethodDef build(BiFunction<VariableDef.This, List<ParameterDef>, StatementDef> bodyBuilder) {
-            StatementDef statement = bodyBuilder.apply(new VariableDef.This(TypeDef.THIS), parameters);
+        public MethodDef build(BiFunction<VariableDef.This, List<VariableDef.MethodParameter>, StatementDef> bodyBuilder) {
+            List<VariableDef.MethodParameter> variables = parameters.stream()
+                .map(ParameterDef::asVariable)
+                .toList();
+            StatementDef statement = bodyBuilder.apply(new VariableDef.This(TypeDef.THIS), variables);
```

Comment:
Out of curiosity, why do we need `ParameterDef` when we have `VariableDef.MethodParameter`? Just by name, they seem to share the same purpose.

Reply:
Essentially we have two APIs - one to define classes, methods, etc and ParameterDef is part of it; the second one is an API to define the code

Reply:
Makes sense now